src/components/TransactionHistoryRows/TransactionGenericRow.tsx

```diff
@@ -69,94 +63,124 @@ export const TransactionGenericRow = ({
   blockTime,
   explorerTxLink,
   compactMode = false,
-  toggleOpen
+  toggleOpen,
+  parentWidth
 }: TransactionGenericRowProps) => {
-  const [isLargerThanMd] = useMediaQuery(`(min-width: ${breakpoints['md']})`)
-  const [isLargerThanLg] = useMediaQuery(`(min-width: ${breakpoints['lg']})`)
-  const [isLargerThanXl] = useMediaQuery(`(min-width: ${breakpoints['xl']})`)
+  const isLargerThanSm = parentWidth > parseInt(breakpoints['sm'], 10)
+  const isLargerThanMd = parentWidth > parseInt(breakpoints['md'], 10)
+  const isLargerThanLg = parentWidth > parseInt(breakpoints['lg'], 10)
+
+  let columns = '1fr'
+  let dateFormat = 'MM/DD/YYYY hh:mm A'
+
+  if (isLargerThanSm) {
+    columns = '1fr 2fr'
+    dateFormat = 'hh:mm A'
+  }
+  if (isLargerThanMd) {
+    columns = '1fr 2fr'
+  }
+  if (isLargerThanLg) {
+    columns = '1fr 2fr 1fr 1fr'
+  }
   return (
-    <Box as='button' w='full' py={4} onClick={() => toggleOpen()}>
-      <Flex width='full' justifyContent='space-between' textAlign='left' alignItems='flex-end'>
-        <Flex alignItems='flex-start' flex={1} mr={3} flexDir='column'>
-          {showDateAndGuide && <TransactionDate blockTime={blockTime} />}
-          <Flex alignItems={{ base: 'flex-end', md: 'center' }} width='full'>
-            <IconCircle mr={3}>
+    <Button
+      height='auto'
+      fontWeight='inherit'
+      variant='unstyled'
+      w='full'
+      p={4}
+      onClick={() => toggleOpen()}
+    >
+      <SimpleGrid
+        gridTemplateColumns={{ base: '1fr', md: columns }}
```

Comment:
If you pass the columns in as an array, chakra will use that value for each of the breakpoints in the theme starting from 0 to largest breakpoint allowing you to remove the checks above.

```tsx
gridTemplateColumns={{ base: '1fr', md: ['1fr 2fr', '1fr 2fr', '1fr 2fr 1fr 1fr'] }}
```

Reply:
i'm not trying to use the viewport size, but the size of the parent container width.

Reply:
The checks above are for the size of the parent container, those media queries will use the viewport size.

Reply:
Sure would be cool if chakra let you override the source of the breakpoint, very edge case but would be neat.